source/dem/pw_contact_force.cc

```diff
@@ -98,5 +100,278 @@ PWContactForce<dim>::update_contact_information(
   contact_info.tangential_relative_velocity = tangential_relative_velocity;
 }
 
+template <int dim>
+Tensor<1, dim>
+  PWContactForce<dim>::calculation_total_torque(Tensor<1, dim> total_force,
+                                                Point<dim>     center_mass,
+                                                Point<dim> point_on_boundary)
+{
+  if (dim == 3)
+    {
+      return cross_product_3d(total_force, point_on_boundary - center_mass);
+    }
+}
```

Comment:
2D implementation is missing? Is it necessary to have a function for this single calculation?

Reply:
I was wondering and skipped. In fact, i think that cross_product_3d works in 2d and yeah, do we need a function. This is more readable with a function i guess